internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -105,40 +101,26 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	// Set the NodeSystemInfo.
 	machine.Status.NodeInfo = &node.Status.NodeInfo
 
-	// Reconcile node annotations.
-	patchHelper, err := patch.NewHelper(node, remoteClient)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	desired := map[string]string{
+	// Compute all the annotations that CAPI is setting on nodes;
+	// CAPI only enforce some annotation, never change or remove them
+	nodeAnnotations := map[string]string{
 		clusterv1.ClusterNameAnnotation:      machine.Spec.ClusterName,
 		clusterv1.ClusterNamespaceAnnotation: machine.GetNamespace(),
 		clusterv1.MachineAnnotation:          machine.Name,
 	}
 	if owner := metav1.GetControllerOfNoCopy(machine); owner != nil {
-		desired[clusterv1.OwnerKindAnnotation] = owner.Kind
-		desired[clusterv1.OwnerNameAnnotation] = owner.Name
-	}
-	if annotations.AddAnnotations(node, desired) {
-		if err := patchHelper.Patch(ctx, node); err != nil {
-			log.V(2).Info("Failed patch node to set annotations", "err", err, "node name", node.Name)
-			return ctrl.Result{}, err
-		}
+		nodeAnnotations[clusterv1.OwnerKindAnnotation] = owner.Kind
+		nodeAnnotations[clusterv1.OwnerNameAnnotation] = owner.Name
 	}
 
-	updatedNode := unstructuredNode(node.Name, node.UID, getManagedLabels(machine.Labels))
-	err = ssa.Patch(ctx, remoteClient, machineManagerName, updatedNode, ssa.WithCachingProxy{Cache: r.ssaCache, Original: node})
-	if err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to apply labels to Node")
-	}
-	// Update `node` with the new version of the object.
-	if err := r.Client.Scheme().Convert(updatedNode, node, nil); err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to convert node to structured object %s", klog.KObj(node))
-	}
+	// Compute labels to be propagated from Machines to nodes.
+	// NOTE: CAPI should manage only a subset of node labels, everything else should be preserved.
+	// NOTE: Once we reconcile node labels for the first time, the NodeUninitializedTaint is removed from the node.
+	nodeLabels := getManagedLabels(machine.Labels)
 
 	// Reconcile node taints
-	if err := r.reconcileNodeTaints(ctx, remoteClient, node); err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile taints on Node %s", klog.KObj(node))
+	if err := r.patchNode(ctx, node, nodeLabels, nodeAnnotations); err != nil {
```

Comment:
```suggestion
	if err := r.patchNode(ctx, remoteClient, node, nodeLabels, nodeAnnotations); err != nil {
```

We need to pass the `remoteClient` to operate on the workload cluster's node.